rclpy/src/rclpy_common/src/common.c

```diff
@@ -34,63 +68,54 @@ rclpy_convert_to_py_qos_policy(void * profile)
     return NULL;
   }
 
+  // start qos setting
   rmw_qos_profile_t * qos_profile = (rmw_qos_profile_t *)profile;
+  rclpy_qos_profile_t rclpy_qos;
+  init_rclpy_qos_profile(&rclpy_qos, pyqos_profile);
 
-  PyObject * pyqos_depth = PyLong_FromSize_t(qos_profile->depth);
-  if (!pyqos_depth) {
-    Py_DECREF(pyqos_profile);
+  int set_result = -1;
+
+  rclpy_qos.depth = PyLong_FromSize_t(qos_profile->depth);
+  if (!rclpy_qos.depth) {
+    cleanup_rclpy_qos_profile(&rclpy_qos, true);
```

Comment:
It's a personal preference, but I think it would be more readable to keep `Py_DECREF(pyqos)` at the same scope level where it is initialized (not in the `rclpy_qos` cleanup function). This would be more in line with pattern we see throughout the rest of the C extension. I don't feel too strongly about it though.

Reply:
I don't feel strongly about it either - changed

Reply:
thanks :)